t/geturl_connectivity.pl

```diff
@@ -84,8 +87,16 @@ sub run_httpd {
         my $name = sprintf("IPv%s client to %s (-%sipv6)",
                            $tc->{client_ipv} || '*', $uri, $tc->{ipv6_opt} ? '' : 'no');
         $ddclient::globals{'ipv6'} = $tc->{ipv6_opt};
+        $ddclient::globals{'curl'} = 0;
         my $got = ddclient::geturl({ url => $uri, ipversion => $tc->{client_ipv} });
         isnt($got // '', '', $name);
+        SKIP: {
+            skip("Curl not available on this system", 1) if !$has_curl;
+            $name = $name . " (curl)";
+            $ddclient::globals{'curl'} = 1;
+            $got = ddclient::geturl({ url => $uri, ipversion => $tc->{client_ipv} });
+            isnt($got // '', '', $name);
+        }
```

Comment:
Add a new `curl` key to the entries in `@test_cases` instead of adding a new test to the loop body. Rationale:
  * It makes it easier to mark a test as TODO if necessary.
  * curl tests should not be skipped if IO::Socket::INET6 is not available.
  * Some cases that don't make sense for IO::Socket do make sense for curl.

Reply:
Yes, well.  That was more work!  I'll look into this.

Reply:
Done.  @rhansen Please check that I got the skips right.